ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2

```diff
@@ -17,6 +17,10 @@ ethernet_interfaces:
 {%             set peer = connected_endpoint %}
 {%             for adapter_switch in adapter.switches %}
 {%                 if adapter_switch == inventory_hostname %}
+{%                     if adapter_settings.monitor_sessions is arista.avd.defined %}
+{%                         do adapter_settings.monitor_sessions.update({'interface': adapter.switch_ports[loop.index0]}) %}
```

Comment:
I think you need to for-loop over the monitor_sessions to update in each entry (if you decide on supporting multiple sessions)

Reply:
I'd prefer to change to "interface**s**": and put the whole list of switch_ports.. then in monitor-sessions.j2 loop over the interfaces list and in conjunction with source or destination decide how to output.
This keeps ethernet-interfaces and port-channel-interfaces leaner..?
 what do you think @ClausHolbechArista ?

Reply:
this will even allow us to use a simple 
```
session.interfaces | join(", ")
```
for both source and destination. Because multiple interfaces in comma sep'ed list is a valid input for eos_cli_config_gen and Arista CLI..

e.g.
DC1-SERVERS
```
  server18_monitoring_session_source:
    rack: RackC
    adapters:
      - endpoint_ports: [ Eth1 ]
        switch_ports: [ Ethernet25, Ethernet26 ]
        switches: [ DC1-SVC3A ]
        profile: TENANT_A
        monitor_sessions:
          name: MyMonitoringSession_server18
          role: source
          source_settings:
            direction: both
            access_group:
              type: ip
              name: MyIpACL
              priority: 5

  server19_monitoring_session_destination:
    rack: RackC
    adapters:
      - endpoint_ports: [ Eth1 ]
        switch_ports: [ Ethernet27, Ethernet28 ]
        switches: [ DC1-SVC3A ]
        monitor_sessions:
          name: MyMonitoringSession_server18
          role: destination
```
structured YAML
```
monitor_sessions:
- name: MyMonitoringSession_server18
  sources:
  - name: Ethernet25, Ethernet26
  destinations:
  - Ethernet27, Ethernet28
```

EOS_CLI
```
monitor session MyMonitoringSession_server18 source Ethernet25, Ethernet26
monitor session MyMonitoringSession_server18 destination Ethernet27, Ethernet28
```

Reply:
that would only work if all settings were the same. We should allow multiple sources with different ACLs or tx/rx/both in the same session. So IMO we should just make the monitor_sessions under connected_endpoints into a list.

Reply:
I totally misunderstood you -.- 
you want different monitoring sessions spawning from the same interfaces. I'll add it